src/state_point.F90

```diff
@@ -214,18 +218,59 @@ subroutine write_state_point()
              group="tallies/tally" // to_str(i), length=t % n_nuclide_bins)
         deallocate(temp_array)
 
-        ! Write number of score bins, score bins, and moment order
+        ! Write number of score bins, score bins, user score bins
         call sp % write_data(t % n_score_bins, "n_score_bins", &
              group="tallies/tally" // to_str(i))
         call sp % write_data(t % score_bins, "score_bins", &
              group="tallies/tally" // to_str(i), length=t % n_score_bins)
-        call sp % write_data(t % moment_order, "moment_order", &
-             group="tallies/tally" // to_str(i), length=t % n_score_bins)
-
-        ! Write number of user score bins
         call sp % write_data(t % n_user_score_bins, "n_user_score_bins", &
              group="tallies/tally" // to_str(i))
 
+      ! Write explicit moment order strings for each score bin
+#ifdef HDF5
+        call sp % open_group("tallies/tally" // trim(to_str(i)) // "/moments")
+        call sp % close_group()
+#endif
+
+        k = 1
+        MOMENT_LOOP: do j = 1, t % n_user_score_bins
+          select case(t % score_bins(k))
+          case (SCORE_SCATTER_N, SCORE_NU_SCATTER_N)
+            moment_name = 'P' // to_str(t % moment_order(k))
+            call sp % write_data(moment_name, "order", &
+                 group="tallies/tally" // trim(to_str(i)) // &
+                 "/moments/" // trim(to_str(k)))
```

Comment:
I understand the motivation to have a separate group for each moment string (output_interface API doesn't provide a way to write an array of strings). It might be better though to just have a separate dataset for each moment string in the tallies/tallyN/moments group, i.e. moments/order1, moments/order2, etc., until output_interface is improved.